### ISSUE TITLE
linter: `no-unused-vars` false positive for `TSTypeAliasDeclaration` with `ExportDefaultDeclaration`

### ISSUE THREAD

**Issue by Sysix** [oxc-project/oxc#7809]
### What version of Oxlint are you using?

0.15.0

### What command did you run?

oxlint

### What does your `.oxlint.json` config file look like?

````
{
  "plugins": ["unicorn", "typescript", "oxc"],
  "categories": {
    "correctness": "error",
    "suspicious": "warn"
  },
  "rules": {
    "no-unused-vars": "error"
  }
}

````

### Setup:

```typescript
/// types.ts:
type ABC = Record<string, unknown>;

export default ABC;

/// index.ts
import Types from './types.ts';

const xxx = (): ABC => ({}};
```

### Result

```
  × eslint(no-unused-vars): Type alias 'ABC' is declared but never used.
    ╭─[src/build-from-oxlint-config/types.ts:21:6]
 20 │ 
 21 │ type ABC = Record<string, unknown>;
    ·      ─┬─
    ·       ╰── 'ABC' is declared here
 22 │ 
    ╰────
  help: Consider removing this declaration.
```

### Expected

No Error

**Comment by Sysix** [comment]
closing because of the implemented test in the pr. thanks ❤️
